azext_hanaonazure/commands.py

```diff
@@ -9,13 +9,16 @@
 from ._client_factory import cf_hanainstance_groups
 from azure.cli.core.commands import CliCommandType
 
+
 def load_command_table(self, _):
     custom_tmpl = 'azext_hanaonazure.custom#{}'
     custom_type = CliCommandType(operations_tmpl=custom_tmpl)
 
     with self.command_group('hanainstance', client_factory=cf_hanainstance_groups) as g:
         g.custom_command('list', 'list_hanainstance')
-        g.custom_command('show', 'show_hanainstance', exception_handler=empty_on_404)
+        g.custom_command('show', 'show_hanainstance',
+                         exception_handler=empty_on_404)
         g.custom_command('restart', 'restart_hanainstance')
+        g.custom_command('enable-monitoring', 'enable_monitoring_hanainstance')
```

Comment:
It seems to me that `monitoring` is a sub-command which have `enable`, `disable` actions and all the monitoring specific parameters to only be valid to this sub-command.

Reply:
Thinking more about the monitoring feature and talking whit few folks around. I strongly believe the command structure should separate the monitoring pieces from the rest of the `hanainstance` ones.

This is the proposal:
`az hanainstance monitor enable` to enable monitoring on the hanainstance
`az hanainstance monitor disable` once the disable part is done, have this command here.
have all the parameters that are only relevant to the monitoring showing only for this sub-command.

Reply:
done